.github/workflows/test.yml

```diff
@@ -88,3 +88,55 @@ jobs:
           git add bricks
           git commit -m "[Chore] Generate Mason bundle"
           git push
+
+  generate_sample_project:
+    # The job generate_sample_project identify that the job generate_mason_bundle must complete successfully
+    # before this generate_sample_project job will run
+    needs: generate_mason_bundle
```

Comment:
Could you please check if we made a change in another part of the core template, such as `pubspec.yaml` instead of `pubspec_dependency.yaml` (generate_mason_bundle job will be skipped), does this workflow run properly to commit the changes to the sample project?